src/test/java/teammates/ui/webapi/CreateNotificationActionTest.java

```diff
@@ -0,0 +1,110 @@
+        return Const.ResourceURIs.NOTIFICATION;
+    }
+
+    @Override
+    String getRequestMethod() {
+        return POST;
+    }
+
+    @Override
+    protected void testExecute() throws Exception {
+        NotificationAttributes notif = typicalBundle.notifications.get("notification1");
+
+        long startTime = notif.getStartTime().toEpochMilli();
+        long endTime = notif.getEndTime().toEpochMilli();
+        NotificationType type = notif.getType();
+        NotificationTargetUser targetUser = notif.getTargetUser();
+        String title = notif.getTitle();
+        String message = notif.getMessage();
+        String invalidTitle = "";
+
+        loginAsAdmin();
+        ______TS("Typical Case: Add notification successfully");
+        CreateNotificationRequest req = buildCreateRequest(startTime, endTime, type, targetUser, title, message);
+        CreateNotificationAction action = getAction(req);
+        NotificationData res = (NotificationData) action.execute().getOutput();
+
+        NotificationAttributes createdNotification = logic.getNotification(res.getNotificationId());
+
+        // check that notification returned has same properties as notification created
+        assertEquals(createdNotification.getStartTime().toEpochMilli(), res.getStartTimestamp());
+        assertEquals(createdNotification.getEndTime().toEpochMilli(), res.getEndTimestamp());
+        assertEquals(createdNotification.getType(), res.getNotificationType());
+        assertEquals(createdNotification.getTargetUser(), res.getTargetUser());
+        assertEquals(createdNotification.getTitle(), res.getTitle());
+        assertEquals(createdNotification.getTitle(), res.getMessage());
+
+        // check DB correctly processed request
+        assertEquals(startTime, createdNotification.getStartTime().toEpochMilli());
+        assertEquals(endTime, createdNotification.getEndTime().toEpochMilli());
+        assertEquals(type, createdNotification.getType());
+        assertEquals(targetUser, createdNotification.getTargetUser());
+        assertEquals(title, createdNotification.getTitle());
+        assertEquals(message, createdNotification.getTitle());
+
+        ______TS("Parameters cannot be null");
+        InvalidHttpRequestBodyException ex = verifyHttpRequestBodyFailure(
+                buildCreateRequest(startTime, endTime, null, targetUser, title, message));
+        assertEquals("Notification type cannot be null", ex.getMessage());
+
+        ex = verifyHttpRequestBodyFailure(buildCreateRequest(startTime, endTime, type, null, title, message));
+        assertEquals("Notification target user cannot be null", ex.getMessage());
+
+        ex = verifyHttpRequestBodyFailure(buildCreateRequest(startTime, endTime, type, targetUser, null, message));
+        assertEquals("Notification title cannot be null", ex.getMessage());
+
+        ex = verifyHttpRequestBodyFailure(buildCreateRequest(startTime, endTime, type, targetUser, title, null));
```

Comment:
Since you are calling this method over and over again and only vary one parameter at a time, it's better to just create a typical request using `getTypicalCreateRequest()` and then set attribute for the request object to `null`.

Reply:
Would that reduce the readability of the code since I would not be able to put it concisely in a single line within  `verifyHttpRequestBodyFailure`, understand that this refactor would be better if we increase the number of test cases.

Reply:
I would actually think readability will improve: 
```
request = getTypicalCreateRequest();
request.setTitle(null);
verifyHttpRequestBodyFailure(request);
```

Notice that now you don't have long lines of code, and it is much easier to tell that you are testing for `null` title